### PR TITLE
feature (refs T15052): remove faulty ORM-Annotations

### PR DESCRIPTION
**Pull request by MoritzMandler**
Ticket:
https://yaits.demos-deutschland.de/T15052

Description: 
remove faulty ORM-Annotations
BLOB and TEXT columns cannot have DEFAULT values.
https://demosdeutschland.slack.com/archives/CEDGU47DX/p1697120716335369

All properties are set with an empty string initially - the default value is set this way.
`private string $property = '';` 

### Linked PRs (optional)
https://github.com/demos-europe/demosplan-core/pull/2145

- [ ] Tests updated/created
- [ ] Update documentation
- [x] Link all relevant tickets
- [ ] Move the tickets on the board accordingly

### CODE REVIEW

File: demosplan/DemosPlanCoreBundle/Entity/Procedure/ProcedureSettings.php [hunk 1]
```diff
@@ -202,28 +202,28 @@ class ProcedureSettings extends CoreEntity implements UuidEntityInterface, Proce
      *
      * @ORM\Column(name="_ps_designated_phase", type="string", length=50, nullable=true)
      */
-    protected $designatedPhase = null;
```
Comment (AlexMenschoid) [inline demosplan/DemosPlanCoreBundle/Entity/Procedure/ProcedureSettings.php:205]: I would not remove the 'null' initialization as it is not a default value in the ORM sense, but might also be needed as a fallback for the values.

File: demosplan/DemosPlanCoreBundle/Entity/Procedure/ProcedureSettings.php [hunk 2]
```diff
@@ -202,28 +202,28 @@ class ProcedureSettings extends CoreEntity implements UuidEntityInterface, Proce
...
+    protected $designatedPhase;
 
     /**
      * @var string
      *
      * @ORM\Column(name="_ps_designated_public_phase", type="string", length=50, nullable=true)
      */
-    protected $designatedPublicPhase = null;
```
Comment (AlexMenschoid) [inline demosplan/DemosPlanCoreBundle/Entity/Procedure/ProcedureSettings.php:212]: see above

File: demosplan/DemosPlanCoreBundle/Entity/Procedure/ProcedureSettings.php [hunk 3]
```diff
@@ -202,28 +202,28 @@ class ProcedureSettings extends CoreEntity implements UuidEntityInterface, Proce
...
+    protected $designatedPublicPhase;
 
     /**
      * @var DateTime
      *
      * @ORM\Column(name="_ps_designated_switch_date", type="datetime", nullable=true)
      */
-    protected $designatedSwitchDate = null;
```
Comment (AlexMenschoid) [inline demosplan/DemosPlanCoreBundle/Entity/Procedure/ProcedureSettings.php:219]: see above

File: demosplan/DemosPlanCoreBundle/Entity/Procedure/ProcedureSettings.php [hunk 4]
```diff
@@ -202,28 +202,28 @@ class ProcedureSettings extends CoreEntity implements UuidEntityInterface, Proce
...
+    protected $designatedSwitchDate;
 
     /**
      * @var DateTime
      *
      * @ORM\Column(name="_ps_designated_public_switch_date", type="datetime", nullable=true)
      */
-    protected $designatedPublicSwitchDate = null;
```
Comment (AlexMenschoid) [inline demosplan/DemosPlanCoreBundle/Entity/Procedure/ProcedureSettings.php:226]: see above

File: demosplan/DemosPlanCoreBundle/Entity/Procedure/ProcedureSettings.php [hunk 5]
```diff
@@ -232,7 +232,7 @@ class ProcedureSettings extends CoreEntity implements UuidEntityInterface, Proce
      *
      * @ORM\JoinColumn(referencedColumnName="_u_id", nullable=true, onDelete="SET NULL")
      */
-    protected $designatedPhaseChangeUser = null;
```
Comment (AlexMenschoid) [inline demosplan/DemosPlanCoreBundle/Entity/Procedure/ProcedureSettings.php:235]: see above

File: demosplan/DemosPlanCoreBundle/Entity/Procedure/ProcedureSettings.php [hunk 6]
```diff
@@ -241,21 +241,21 @@ class ProcedureSettings extends CoreEntity implements UuidEntityInterface, Proce
      *
      * @ORM\JoinColumn(referencedColumnName="_u_id", nullable=true, onDelete="SET NULL")
      */
-    protected $designatedPublicPhaseChangeUser = null;
```
Comment (AlexMenschoid) [inline demosplan/DemosPlanCoreBundle/Entity/Procedure/ProcedureSettings.php:244]: see above

File: demosplan/DemosPlanCoreBundle/Entity/Procedure/ProcedureSettings.php [hunk 7]
```diff
@@ -241,21 +241,21 @@ class ProcedureSettings extends CoreEntity implements UuidEntityInterface, Proce
...
+    protected $designatedPublicPhaseChangeUser;
 
     /**
      * @var DateTime
      *
      * @ORM\Column(name="_ps_designated_end_date", type="datetime", nullable=true)
      */
-    protected $designatedEndDate = null;
```
Comment (AlexMenschoid) [inline demosplan/DemosPlanCoreBundle/Entity/Procedure/ProcedureSettings.php:251]: see above

File: demosplan/DemosPlanCoreBundle/Entity/Procedure/ProcedureSettings.php [hunk 8]
```diff
@@ -241,21 +241,21 @@ class ProcedureSettings extends CoreEntity implements UuidEntityInterface, Proce
...
+    protected $designatedEndDate;
 
     /**
      * @var DateTime
      *
      * @ORM\Column(name="_ps_designated_public_end_date", type="datetime", nullable=true)
      */
-    protected $designatedPublicEndDate = null;
```
Comment (AlexMenschoid) [inline demosplan/DemosPlanCoreBundle/Entity/Procedure/ProcedureSettings.php:258]: see above

File: demosplan/DemosPlanCoreBundle/Entity/Procedure/ProcedureSettings.php [hunk 9]
```diff
@@ -902,8 +900,6 @@ public function getDesignatedPublicPhase()
 
     /**
      * @param string $designatedPublicPhase
-     *
-     * @return mixed
```
Comment (AlexMenschoid) [inline demosplan/DemosPlanCoreBundle/Entity/Procedure/ProcedureSettings.php:906]: if you remove the docblock type hint, then pls add it as an actual type hint. Which would be, in fact, better. :)

File: demosplan/DemosPlanCoreBundle/Entity/Procedure/ProcedureSettings.php [hunk 10]
```diff
@@ -977,8 +973,6 @@ public function getDesignatedPublicEndDate(): ?DateTime
     }
 
     /**
-     * @param mixed $designatedPublicEndDate
-     *
```
Comment (AlexMenschoid) [inline demosplan/DemosPlanCoreBundle/Entity/Procedure/ProcedureSettings.php:981]: see above

File: demosplan/DemosPlanCoreBundle/Entity/User/Customer.php [hunk 11]
```diff
@@ -186,18 +186,18 @@ class Customer extends CoreEntity implements UuidEntityInterface, CustomerInterf
...
 
     public function __construct(/**
      * @ORM\Column(name="_c_name", type="string", length=50, nullable=false)
      */
-    private string $name, /**
+        private string $name, /**
      * @ORM\Column(name="_c_subdomain", type="string", length=50, nullable=false)
      */
-    private string $subdomain, string $mapAttribution = '')
+        private string $subdomain, string $mapAttribution = '')
```
Comment (AlexMenschoid) [inline demosplan/DemosPlanCoreBundle/Entity/User/Customer.php:200]: This indentation seems to be a mistake.

**Review by AlexMenschoid — CHANGES_REQUESTED** [review]

File: demosplan/DemosPlanCoreBundle/Entity/Procedure/ProcedureSettings.php (hunk 2 above)
Comment (MoritzMandler) [reply demosplan/DemosPlanCoreBundle/Entity/Procedure/ProcedureSettings.php:212]: I dont like what the style fixer does!
It wasn't me 4d62cca1cbf0553c05be7f209bee90ba2e429965

File: demosplan/DemosPlanCoreBundle/Entity/Procedure/ProcedureSettings.php (hunk 3 above)
Comment (MoritzMandler) [reply demosplan/DemosPlanCoreBundle/Entity/Procedure/ProcedureSettings.php:219]: I dont like what the style fixer does!
It wasn't me 4d62cca1cbf0553c05be7f209bee90ba2e429965

File: demosplan/DemosPlanCoreBundle/Entity/Procedure/ProcedureSettings.php (hunk 9 above)
Comment (MoritzMandler) [reply demosplan/DemosPlanCoreBundle/Entity/Procedure/ProcedureSettings.php:906]: I dont like what the style fixer does!
It wasn't me 4d62cca1cbf0553c05be7f209bee90ba2e429965

File: demosplan/DemosPlanCoreBundle/Entity/User/Customer.php (hunk 11 above)
Comment (MoritzMandler) [reply demosplan/DemosPlanCoreBundle/Entity/User/Customer.php:200]: I dont like what the style fixer does!
It wasn't me 4d62cca1cbf0553c05be7f209bee90ba2e429965

**Review by AlexMenschoid — APPROVED** [review]
as all concerns were the product of the style fixer, they can be ignored. It means we should readjust our style fixer, but that is not the scope of this PR.

**Pull request merged**